### PR TITLE
Add typehints only if Entity class exist

### PR DESCRIPTION
**Pull request by dmromanov**
In addition to a pull #337

### CODE REVIEW

File: src/Template/Bake/Template/index.ctp [hunk 1]
```diff
@@ -16,7 +16,9 @@
 <?php
 /**
   * @var \<%= $namespace %>\View\AppView $this
+<% if ($entityClass): %>
```
Comment (ndm2) [inline src/Template/Bake/Template/index.ctp]: Could not defining a view variable at all maybe be a little to hard of a cut? How about falling back to `\Cake\ORM\Entity`?

Comment (dereuromark) [inline src/Template/Bake/Template/index.ctp]: The entity interface makes total sense 👍 
Once the actual entity exists, the IdeHelper will replace the annotation then automatically anyway.

Comment (markstory) [inline src/Template/Bake/Template/index.ctp]: :+1: to falling back to EntityInterface.

File: tests/comparisons/Template/testBakeIndex.ctp [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?php
 /**
   * @var \Bake\Test\App\View\AppView $this
-  * @var \Bake\Test\App\Model\Entity\TemplateTaskComment[]|\Cake\Collection\CollectionInterface $templateTaskComments
+  * @var \Bake\Test\App\Model\Entity\Cake\Datasource\EntityInterface[]|\Cake\Collection\CollectionInterface $templateTaskComments
```
Comment (ndm2) [inline tests/comparisons/Template/testBakeIndex.ctp]: That looks a little weird, the interface name shouldn't be prefixed with `\Bake\Test\App\Model\Entity`.

Comment (dereuromark) [inline tests/comparisons/Template/testBakeIndex.ctp]: Oh you are right

Comment (dmromanov) [inline tests/comparisons/Template/testBakeIndex.ctp]: Will fix.

**Pull request merged**